client/src/pages/auth/Login.tsx

```diff
@@ -1,7 +1,67 @@
 import React, { ReactElement } from 'react';
+import { useFormik } from 'formik';
+import * as Yup from 'yup';
+import Button from '@mui/material/Button';
+import TextField from '@mui/material/TextField';
+import { firebase } from '../../services/firebase';
 
 function Login(): ReactElement {
-  return <div>Login component</div>;
+  const formik = useFormik({
+    initialValues: {
+      email: '',
+      password: '',
+    },
+    validationSchema: Yup.object({
+      email: Yup.string().email('Please enter a valid email address.').required('Required.'),
+      password: Yup.string().required('Required.').min(8, 'At least 8 characters'),
+    }),
+    validateOnChange: true,
+    onSubmit: async (values) => {
+      // todo add errors handler
+      await firebase.auth.signInWithEmailAndPassword(values.email, values.password);
+      // todo add toast component with success message
+    },
+  });
```

Comment:
isn't it a better practice to extract the validation to a separate file?

Reply:
mostly yes but in this case we are using login only once like register so logic could be placed in component